src/editor.rs

```diff
@@ -655,13 +656,14 @@ impl Editor {
     ///
     /// Will Return `Err` if any error occur.
     pub fn run(&mut self, file_name: Option<String>) -> Result<(), Error> {
-        if let Some(path) = file_name.as_ref().map(Path::new) {
-            self.select_syntax_highlight(path)?;
-            self.load(path)?;
+        if let Some(path) = file_name.as_ref().map(sys::path) {
+            self.select_syntax_highlight(path.as_path())?;
+            self.load(path.as_path())?;
+            self.file_name = Some(path.to_string_lossy().to_string());
```

Comment:
This is needed because on WASI it is using the relative paths to adjust the path which needs to be recorded so that it saves correctly. the alternative is that it invokes the function again on save but i didnt want to change your code too much so i just saved the result on the relative path resolution instead.